controllers/actions.github.com/autoscalinglistener_controller.go

```diff
@@ -345,7 +346,24 @@ func (r *AutoscalingListenerReconciler) createServiceAccountForListener(ctx cont
 }
 
 func (r *AutoscalingListenerReconciler) createListenerPod(ctx context.Context, autoscalingRunnerSet *v1alpha1.AutoscalingRunnerSet, autoscalingListener *v1alpha1.AutoscalingListener, serviceAccount *corev1.ServiceAccount, secret *corev1.Secret, logger logr.Logger) (ctrl.Result, error) {
-	newPod := r.resourceBuilder.newScaleSetListenerPod(autoscalingListener, serviceAccount, secret)
+	var envs []corev1.EnvVar
+	if autoscalingListener.Spec.Proxy != nil {
+		proxyEnv, err := autoscalingListener.Spec.Proxy.EnvVars(func(s string) (*corev1.Secret, error) {
+			var secret corev1.Secret
+			err := r.Get(ctx, types.NamespacedName{Name: s, Namespace: autoscalingRunnerSet.Namespace}, &secret)
+			if err != nil {
+				return nil, err
```

Comment:
should we pass the logger into these places so we can log extra info about each error?

Reply:
I've added logging in place where we have a logger already, otherwise I've checked the other instances: the error returned is bubbled up and logged immediately from the caller